# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 542e982e7537
# Full Hash: 542e982e753711fa0ac5c55da967cb6cbd34e6d5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 4
# Description:
#   Bug 1654112 - Avoid locking the conduit lock when starting/stopping/reconfiguring streams, and clean up. r=bwc
#   
#   A deadlock may happen when main thread calls into a conduit's
#   GetUpstreamRtpSources() as it grabs the conduit lock, while at the same time a
#   platform codec may be blocked on a sync runnable to main in a stream start()
# ==============================================================================

diff -r c01614dfe8f6 -r 542e982e7537 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Sep 01 12:03:36 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Sep 01 15:26:49 2021 +0200
@@ -84,10 +84,10 @@
   Maybe<DOMHighResTimeStamp> LastRtcpReceived() const override;
   DOMHighResTimeStamp GetNow() const override;
 
-  MediaConduitErrorCode StopTransmittingLocked();
-  MediaConduitErrorCode StartTransmittingLocked();
-  MediaConduitErrorCode StopReceivingLocked();
-  MediaConduitErrorCode StartReceivingLocked();
+  void StopTransmitting();
+  void StartTransmitting();
+  void StopReceiving();
+  void StartReceiving();
 
   /**
    * Function to select and change the encoding resolution based on incoming
@@ -161,8 +161,8 @@
   bool GetRemoteSSRC(uint32_t* ssrc) const override;
 
   void UnsetRemoteSSRC(uint32_t ssrc) override;
-  bool SetRemoteSSRCLocked(uint32_t ssrc, uint32_t rtxSsrc);
-
+  void SetRemoteSSRCConfig(uint32_t ssrc, uint32_t rtxSsrc);
+  void SetRemoteSSRCAndRestartAsNeeded(uint32_t ssrc, uint32_t rtxSsrc);
   // Creating a recv stream or a send stream requires a local ssrc to be
   // configured. This method will generate one if needed.
   void EnsureLocalSSRC();
@@ -253,9 +253,9 @@
   // Video Latency Test averaging filter
   void VideoLatencyUpdate(uint64_t new_sample);
 
-  MediaConduitErrorCode CreateSendStream();
+  void CreateSendStream();
   void DeleteSendStream();
-  MediaConduitErrorCode CreateRecvStream();
+  void CreateRecvStream();
   void DeleteRecvStream();
 
   void DeliverPacket(rtc::CopyOnWriteBuffer packet, PacketType type) override;
